### PR TITLE
QueryEditor: Clean-up interface to only have one PanelData

### LINKED ISSUE #18837: QueryEditor interface has panelData and queryResponse (also panel data type)
Think it should only have one of these.

### PR DESCRIPTION
**Pull request by torkelo**
Fixes #18837

QueryEditorProps had both panelData and queryResponse props (both of type PanelData). Now only have queryResponse.

### CODE REVIEW

File: packages/grafana-ui/src/types/datasource.ts [hunk 1]
```diff
@@ -279,8 +279,10 @@ export interface QueryEditorProps<
   query: TQuery;
   onRunQuery: () => void;
   onChange: (value: TQuery) => void;
-  panelData: PanelData; // The current panel data
-  queryResponse?: PanelData; // data filtered to only this query.  Includes the error.
+  /*
+   * Contains query response filtered by refId and possible query error
+   */
+  queryResponse?: PanelData;
```
Comment (torkelo) [inline packages/grafana-ui/src/types/datasource.ts]: don't know what the best name for this prop is. 

* data (like i PanelProps)
* queryResult
* panelData
* queryResponse

Comment (ryantxu) [reply packages/grafana-ui/src/types/datasource.ts]: data 👍

Comment (ryantxu) [reply packages/grafana-ui/src/types/datasource.ts]: The only counter to that is if `series` should really be called `data`... then we end up with data.data.   
```
export interface PanelData {
  state: LoadingState;
  series: DataFrame[];
  request?: DataQueryRequest;
  error?: DataQueryError;
}
```

Or we live with "series" as an akward naming artifact.  (I'm OK with this)

Comment (torkelo) [reply packages/grafana-ui/src/types/datasource.ts]: well we can always rename series to frames. but maybe it's fine as series. Also ok with that.

**Pull request merged**